### PR TITLE
Update free-courses-en.md

### PR DESCRIPTION
**Pull request by ak191201**
Added System Design course

## What does this PR do?
Add resource(s) 

### Description
The most popular guide to cracking system design interviews.

### Why is this valuable (or not)?
This course is the most preferred one when it comes to system design interviews. The course contains all the fundamental and advanced concepts. It also contains real life practical examples.

### How do we know it's really free?
The course is available as a playlist on YouTube. So it's free.

### For book lists, is it a book? For course lists, is it a course? etc.
It's an online course available on YouTube.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -347,6 +347,7 @@
 * [MIT's Introduction to Computer Science and Programming](https://ocw.mit.edu/courses/electrical-engineering-and-computer-science/6-00sc-introduction-to-computer-science-and-programming-spring-2011/) - John Guttag (MIT OpenCourseWare)
 * [MIT's Introduction to Computer Science and Programming in Python](https://ocw.mit.edu/courses/electrical-engineering-and-computer-science/6-0001-introduction-to-computer-science-and-programming-in-python-fall-2016/) - Ana Bell, Eric Grimson, John Guttag (MIT OpenCourseWare)
 * [MIT's Mathematics for Computer Science](http://ocw.mit.edu/courses/electrical-engineering-and-computer-science/6-042j-mathematics-for-computer-science-fall-2010/video-lectures/)
+* [System Design](https://www.youtube.com/playlist?list=PLMCXHnjXnTnvo6alSjVkgxV-VH6EPyvoX) - Gaurav Sen
```
Comment (shrn01) [inline courses/free-courses-en.md:350]: Looks like this is already added in the Hindi file. But this is an english course. Can you remove it from that file?

Comment (ak191201) [reply courses/free-courses-en.md:350]: I have removed that course from the Hindi file

**Review by shrn01 — APPROVED** [review]
LGTM

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**